### DISCUSSION TITLE
If a connection is opened a little while after the first and they are multiplexed then it fails

### DISCUSSION THREAD

**Discussion by lastlambda** [miguelgrinberg/Flask-SocketIO#2011] [Q&A] [answered]
I was unsure wether to raise this as an issue but I hope that someone is able to help. Essentially if I connect to a namespace (say `/control`) and then connect to a second (say `/test`) a little while later (even as little as 10ms) the second connection will fail with the generic message "Unable to connect".

Attatched is a screenshot of the brief transmission and failure.
![image_2023-09-23_182946048](https://github.com/miguelgrinberg/Flask-SocketIO/assets/67469814/edecfb79-95fa-478d-b5a7-9622f640c295)

I beleive this is unwanted behaviour as I set up the same thing with a node JS socket IO server and it did not fail.

I would like this behaviour as a I have a `control` namespace which can signal for some modules in my website to be enabled. These modules then communicate on a different namesapce for clarities sake.

A solution is to disable multiplexing but that seams to me to be inelegant as you are adding the overhead of mainting a second (up to 3 in my use case) new connections. The other option I have is to simply set up all the connections at the very start and then just not use the ones I dont end up needing. However, (again for clarity) I would like to make use of the connect event because I can reject invalid connections at this time rather than checking for every incoming event.

### To recreate this I used the following files:
For Flask server:
```
## app.py ##

from flask import Flask, render_template
from flask_socketio import SocketIO, emit
    
app = Flask(__name__)
app.config['SECRET_KEY'] = 'secret!'
socketio = SocketIO(app)

@app.route('/')
def index():
    return render_template('index.html')

@socketio.on('connect', namespace="/test1")
def connect():
    print("a")

@socketio.on('connect', namespace="/test2")
def connect():
    print("b")

if __name__ == '__main__':
    socketio.run(app)
```

```
<!--index.html-->
<script src="https://cdn.socket.io/4.6.0/socket.io.min.js" integrity="sha384-c79GN5VsunZvi+Q/WObgk2in0CbZsHnjEqvFxC5DxHn9lTfNce2WW6h2pH6u/kF+" crossorigin="anonymous"></script>
<script>
	let socket = io("/test1")
	setTimeout(function() {
		let socket = io("/test2")
	}, 10);
</script>
```

For Node Js server:
```
// index.js
const express = require('express');
const app = express();
const http = require('http');
const server = http.createServer(app);
const { Server } = require("socket.io");
const io = new Server(server);

app.get('/', (req, res) => {
  res.sendFile(__dirname + '/index.html');
});

io.of('/test1').on('connection', (socket) => {
  console.log('a');
});

io.of('/test2').on('connection', (socket) => {
  console.log('b');
});


server.listen(3000, () => {
  console.log('listening on *:3000');
});
```

**Comment by miguelgrinberg** [comment, marked as answer]
Did you notice in your WebSocket screen capture that the client sent two connection requests for `/test2`? These are the lines that with `40/test2,`.

The server responded with a successful connection to the first of them. This is the line that reads `40/test2,{"sid":"..."}`. It responded with a `Unable to connect` error to the second connection request, because you can't have two connections to the same namespace on the same Engine.IO transport.

I don't know why the client is sending two connections. This must have been a bug in the 4.6.0 version that you are using. The 4.7.2 release does not seem to do this.

**Reply by lastlambda** [reply]
Thank you for your quick reply
I did notice that, I spent a long time going down a rabbit hole of trying to figure out why their where two, I will update and let you know if that fixes

**Reply by lastlambda** [reply]
Thank you that works perfectly :)
